internal/pkg/cli/pipeline_init.go

```diff
@@ -374,24 +494,54 @@ func (o *initPipelineOpts) selectURL() error {
 // bbhttps	https://huanjani@bitbucket.org/huanjani/aws-copilot-sample-service.git (fetch)
 // bbssh	ssh://git@bitbucket.org:teamsinspace/documentation-tests.git (fetch)
 
-// parseGitRemoteResults returns just the trimmed middle column (url) of the `git remote -v` results,
-// and skips urls from unsupported sources.
-func (o *initPipelineOpts) parseGitRemoteResult(s string) ([]string, error) {
-	var urls []string
-	urlSet := make(map[string]bool)
+// parseGitRemoteResults returns just the first (remote name) and second (url) columns of the `git remote -v` results as a map (url: name), and skips urls from unsupported sources.
+func (o *initPipelineOpts) parseGitRemoteResult(s string) []repoResult {
+	var repos []repoResult
 	items := strings.Split(s, "\n")
 	for _, item := range items {
 		if !strings.Contains(item, githubURL) && !strings.Contains(item, ccIdentifier) && !strings.Contains(item, bbURL) {
 			continue
 		}
 		cols := strings.Split(item, "\t")
-		url := strings.TrimSpace(strings.TrimSuffix(strings.Split(cols[1], " ")[0], ".git"))
-		urlSet[url] = true
+		URL := strings.TrimSpace(strings.Split(cols[1], " ")[0])
+		repos = append (repos, repoResult{
+			remoteName: cols[0],
+			remoteURL: URL,
+		})
+	}
+	return repos
+}
+
+// parseGitBranchResults takes the output from `git branch -a -l '[remoteName]/*'`
+// ex:  remotes/cc/main
+//  	remotes/cc/dev
+// and returns just the branch names
+// ex: []string{"main", "dev"}
+func (o *initPipelineOpts) parseGitBranchResults(s string) ([]string, error) {
+	var branches []string
+	items := strings.Split(s, "\n")
+	for _, item := range items {
+		elements := strings.Split(item, "/")
+		if len(elements) < 3 {
+			return nil, fmt.Errorf("cannot parse branch name from '%s'", item)
+		}
+		branchName := elements[2]
+		branches = append(branches, branchName)
+	}
+	return branches, nil
+}
+
+func (o *initPipelineOpts) getRemoteNameFromURL() (string, error) {
+	repos, err := o.fetchAndParseURLs()
+	if err != nil {
+		return "", err
 	}
-	for url := range urlSet {
-		urls = append(urls, url)
+	for _, repo := range repos {
+		if repo.remoteURL == o.repoURL {
+			return repo.remoteName, nil
+		}
 	}
-	return urls, nil
+	return "", fmt.Errorf("URL '%s' is not a local git remote", o.repoURL)
```

Comment:
I wonder if it's syntactically correct to say `fmt.Errorf("url '%s' is not a local git remote", o.repoURL)`?